### PR TITLE
DBF Energy fluctuations

### PR DESCRIPTION
**Pull request by Edoardo-Pedicillo**
Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

**Review by andrea-pasquale — COMMENTED** [review]
Thanks @Edoardo-Pedicillo for implementing this.
Generally it looks good to me. Only a small comment regarding a fix for GPU.
Moreover, feel free to add a small test so that we don't lose coverage.

File: tests/test_models_dbf.py [hunk 1]
```diff
@@ -51,3 +52,11 @@ def test_double_bracket_flow_single_commutator(backend, nqubits):
         dbf(mode=FlowGeneratorType.single_commutator, step=0.01, d=d)
 
     assert initial_off_diagonal_norm > dbf.off_diagonal_norm
+
+
+def test_energy_fluctuations(backend):
+    h0 = np.array([[1, 0], [0, -1]])
+    state = np.array([1, 0])
+    dbf = DoubleBracketFlow(Hamiltonian(1, matrix=h0))
```
Comment (andrea-pasquale) [inline tests/test_models_dbf.py:60]:
```suggestion
    dbf = DoubleBracketFlow(Hamiltonian(1, matrix=h0, backend=backend))
```

**Review by andrea-pasquale — COMMENTED** [review]
Thanks @Edoardo-Pedicillo for adding tests.
Two quick fixes for GPU.

**Review by andrea-pasquale — APPROVED** [review]

**Review by MatteoRobbiati — APPROVED** [review]
Thanks, looks good to me.

**Pull request merged**